src/main/java/com/glodblock/github/common/tile/TileLevelMaintainer.java

```diff
@@ -141,6 +142,14 @@ private TickRateModulation doWork() {
             final IGrid grid = getProxy().getGrid();
             final IItemList<IAEItemStack> inv = getProxy().getStorage().getItemInventory().getStorageList();
 
+            // Check there are available crafting CPUs before doing any work.
+            // This hopefully stops level maintainers busy-looping calculating
+            // crafting tasks that cannot be successfully submitted.
+            boolean allBusy = true;
+            for (final ICraftingCPU cpu : craftingGrid.getCpus()) {
+                allBusy = allBusy && cpu.isBusy();
```

Comment:
Minor nitpick: break loop earlier
```suggestion
                if (!cpu.isBusy()) {
                    allBusy = false;
                    break;
                }
```

Reply:
I was tempted to wax lyrical about the relative performance of branch-per-loop-iteration vs. potentially-unrollable-and-vectorizable-boolean-operations then I remembered the JVM was likely to make this moot ;-)